### PR TITLE
Update CAPT GIT_TAG and go version

### PR DESCRIPTION
**Pull request by panktishah26**
*Description of changes:*

This PR includes,
- Updates GIT_TAG version of CAPT to the latest commit
- Removes patches related to previous CAPT version
- Bumps of Go version to 1.17 to match CAPT's latest version
- Updates checksum

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG [hunk 1]
```diff
@@ -1 +1 @@
-v0.1.0
+77df39b0a63a28deb91030e6b2cd88b5cb75286a
```
Comment (jaxesn) [inline projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG:1]: Tags would be better. Are yall working with them to cut an actually release/tag at some point?

**Review by jaxesn — COMMENTED** [review]
All these patches were upstreamed?

**Comment by pokearu** [comment]
Are we sure we don't need any of these patches? Or are we going to add the ones we need on later PRs?

File: projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG (hunk 1 above)
Comment (panktishah26) [reply projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG:1]: Regarding patches, 
- Most of the patches were PBNJ related which we don't need now. Aravind will work on adding new Rufio patches to CAPT after he implements it
- Few of them are upstreamed or need modification which we will be doing upstream only.

I am not aware of cutting a new release/tag, I will confirm with them.

**Comment by panktishah26** [comment]
> All these patches were upstreamed?



> Are we sure we don't need any of these patches? Or are we going to add the ones we need on later PRs?

Yeah, we would add the patches in future if we need. We don't need any patches as of now.

File: projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG (hunk 1 above)
Comment (abhinavmpandey08) [reply projects/tinkerbell/cluster-api-provider-tinkerbell/GIT_TAG:1]: Yep we will cut a tag once all the changes are finalized. For now, we are just tracking the commits to test the kubified components

**Review by abhinavmpandey08 — APPROVED** [review]
/approve

**Pull request merged**